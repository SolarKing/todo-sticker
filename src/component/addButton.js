// @flow
import React from 'react';
import Translate from '../class/translate';
import AddModal from '../container/addTodoModalContainer';
import styled from 'styled-components';

type Props = {
}

type States = {
    showEditModal: boolean,
    defaultTimestamp: number
}

export default class AddButton extends React.Component<Props, States> {
    handleOpenModal: Function;
    handleCloseModal: Function;
    handleOkModal: Function;

    state = {
        showEditModal: false,
        defaultTimestamp: new Date().getTime()
    };

    constructor(props: Props) {
        super(props);
        this.handleOpenModal = this.handleOpenModal.bind(this);
        this.handleCloseModal = this.handleCloseModal.bind(this);
        this.handleOkModal = this.handleOkModal.bind(this);
    }

    handleOkModal() {
        this.setState({ showEditModal: false });
    }

    handleOpenModal() {
        this.setState({
            showEditModal: true,
            defaultTimestamp: new Date().getTime()
        });
    }

    handleCloseModal() {
        this.setState({ showEditModal: false });
    }

    render() {

        return (
            <div {...this.props}>
                <AddButtonIcon onClick={this.handleOpenModal}>
                    <span className="glyphicon glyphicon-plus-sign"></span>
                </AddButtonIcon>
                <AddModal
                    onSave={this.handleOkModal}
                    onCancel={this.handleCloseModal}
                    title={Translate.tr('Add Event')}
                    visible={this.state.showEditModal}
                    defaultDueDatetime={this.state.defaultTimestamp}
                    defaultDescription={''}
                />
            </div>
        );
    }
}

const AddButtonIcon = styled.div`
    font-size: 38px !important;
    cursor: pointer;
    
    &:hover {
        color: #6494c4;
    }
`;
AddButtonIcon.displayName = 'AddButtonIcon';