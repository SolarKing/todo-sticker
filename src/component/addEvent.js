// @flow
import React from 'react';
import StringTool from 'string';

type Props = {
    handleAddEvent: (descrtiption: string) => void
}

type State = {
    isCanAdd: boolean,
};


export default class AddEvent extends React.Component<Props, State> {
    textarea: ?HTMLTextAreaElement;
    handleTextareaChange: Function;
    descrtiption: string;

    constructor(props: Props) {
        super(props);
        this.state = {
            isCanAdd: false
        };
        this.handleTextareaChange = this.handleTextareaChange.bind(this);
    }

    handleTextareaChange(event: SyntheticEvent<HTMLTextAreaElement>) {

        this.descrtiption = event.currentTarget.value;

        if (this.descrtiption == null || StringTool(this.descrtiption).isEmpty()) {
            this.setState({isCanAdd: false});
            return;
        }
        this.setState({isCanAdd: true});
    }

    render() {
        return (
            <div>
                <textarea onChange={this.handleTextareaChange} />
                <br/>
                <button disabled={!this.state.isCanAdd} onClick={() => this.props.handleAddEvent(this.descrtiption)}> 12 </button>
            </div>
        );
    }
}

