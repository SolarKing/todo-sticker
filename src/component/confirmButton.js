// @flow
import React, { type Node } from 'react';
import { Modal } from 'antd';

type Config = {
    title: string,
    okText: string,
    cancelText: string,
    onOk: () => void,
}

type Props = {
    config: Config,
    children?: Node,
}

export default class ConfirmButton extends React.Component<Props> {

    constructor(props: Props) {
        super(props);
    }

    _showConfirm() {
        Modal.confirm(this.props.config);
    }

    render() {
        let {
            config,
            children,
            ...others
        } = this.props;

        return (
            <a {...others} onClick={this._showConfirm.bind(this)} >{children}</a>
        );
    }
}
