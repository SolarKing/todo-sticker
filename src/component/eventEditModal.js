// @flow
import { Modal, DatePicker, TimePicker, Input } from 'antd';
import React from 'react';
import ReactDOM from 'react-dom';
import moment from 'moment';
import Translate from '../class/translate';
import type {TodoState} from '../states';

type Props = {
    title: string,
    todoState?: TodoState,
    onSave: (newTodoState: TodoState) => void,
}

type States = {
    todoState: TodoState,
    visible: boolean,
}

export default class EditModal extends React.Component<Props, States> {

    static defaultProps = {
        visible: false,
        defaultDescriptin: '',
    }

    constructor(props: Props) {
        super(props);

        let defaultDate = new Date();
        defaultDate.setUTCSeconds(0);

        const {
            todoState = {
                dueDatetime: defaultDate.getTime(),
                description: ''
            }
        } = this.props;

        this.state = {
            todoState,
            visible: true,
        };
    }

    _handleDateChanged(dates: moment) {
        let {
            todoState
        } = this.state;

        let dueDatetime = moment(todoState.dueDatetime);
        dueDatetime.set('year', dates.get('year'));
        dueDatetime.set('month', dates.get('month'));
        dueDatetime.set('date', dates.get('date'));
        todoState.dueDatetime = dueDatetime.toDate().getTime();

        this.setState({
            todoState
        });
    }

    _handleTimeChanged(time: moment) {
        let {
            todoState
        } = this.state;

        let dueDatetime = moment(todoState.dueDatetime);
        dueDatetime.set('hour', time.get('hour'));
        dueDatetime.set('minute', time.get('minute'));
        todoState.dueDatetime = dueDatetime.toDate().getTime();

        this.setState({
            todoState
        });
    }

    _handleDescriptionChanged(event: SyntheticEvent<HTMLTextAreaElement>) {
        let {
            todoState
        } = this.state;
        todoState.description = event.currentTarget.value;

        this.setState({
            todoState
        });
    }

    _handleOk() {
        this.props.onSave(this.state.todoState);
        this.setState({
            visible: false,
        });
    }

    _handleCancel() {
        this.setState({
            visible: false,
        });
    }

    render() {
        const {
            onSave,
            title,
            // onClose,
            ...others
        } = this.props;

        const {
            todoState
        } = this.state;

        return (
            <Modal
                {...others}
                title={title}
                destroyOnClose={true}
                onOk={this._handleOk.bind(this)}
                onCancel={this._handleCancel.bind(this)}
                visible={this.state.visible}
            >
                <p>{Translate.tr('Due date')}</p>
                <DatePicker 
                    allowClear={false}
                    onChange={this._handleDateChanged.bind(this)}
                    defaultValue={moment(todoState.dueDatetime)}
                />
                &nbsp;
                <TimePicker
                    allowEmpty={false}
                    onChange={this._handleTimeChanged.bind(this)}
                    defaultValue={moment(todoState.dueDatetime)}
                    format={'HH:mm'}
                />
                <p>{Translate.tr('eventDescription')}</p>
                <Input.TextArea
                    defaultValue={todoState.description}
                    onChange={this._handleDescriptionChanged.bind(this)}
                    rows={4}
                />
            </Modal>
        );
    }
}

export const TodoEditableModal = {
    show(props: Props) {
        let wrapper = document.getElementById('todoModal');
        if (!wrapper) {
            return;
        }

        let cleanup = () => {
            ReactDOM.unmountComponentAtNode(wrapper);
        };

        ReactDOM.render(
            <EditModal
                {...props}
                afterClose={cleanup}
            />
            , wrapper
        );
    }
};