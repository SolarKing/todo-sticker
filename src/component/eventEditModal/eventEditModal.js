// @flow
import { Modal, DatePicker, TimePicker, Input } from 'antd';
import React from 'react';
import moment from 'moment';
import 'moment/locale/zh-cn';
import Translate from '../../class/translate';

export type EditInfo = {
    date: number,
    time: number,
    desctiption: string
}

type Props = {
    visible: boolean,
    title: string,
    defaultDate?: number,
    defaultTime?: number,
    defaultDescriptin?: string,
    onSave?: (editInfo: EditInfo) => void
}

type States = {
    date: number,
    time: number,
    desctiption: string
}

export default class EditModal extends React.Component<Props, States> {

    handleDateChanged: Function;
    handleTimeChanged: Function;
    handleDescriptionChanged: Function;
    handleOk: Function;

    static defaultProps = {
        visible: false,
        defaultDate: new Date().getTime(),
        defaultTime: new Date().getTime(),
        defaultDescriptin: '',
    }

    state = {
        date: new Date().getTime(),
        time: new Date().getTime(),
        desctiption: ''
    };

    constructor(props: Props) {
        super(props);
        moment.locale('en');
        this.handleDateChanged = this.handleDateChanged.bind(this);
        this.handleTimeChanged = this.handleTimeChanged.bind(this);
        this.handleDescriptionChanged = this.handleDescriptionChanged.bind(this);
        this.handleOk = this.handleOk.bind(this);
    }

    handleDateChanged(dates: moment) {
        this.setState(
            {
                date: dates
            }
        );
    }

    handleTimeChanged(time: moment) {
        this.setState(
            {
                time: time
            }
        );
    }

    handleDescriptionChanged(event: SyntheticEvent<HTMLTextAreaElement>) {
        let descriptin = event.currentTarget.value;
        this.setState({
            desctiption: descriptin
        });
    }

    handleOk() {
        if (!this.props.onSave) {
            return;
        }
        let editInfo: EditInfo;
        editInfo = {
            date: this.state.date,
            time: this.state.time,
            desctiption: this.state.desctiption
        };
        this.props.onSave(editInfo);
    }

    render() {
        var {
            defaultDate,
            defaultTime,
            defaultDescriptin,
            ...others
        } = this.props;

        return (
            <Modal onOk={this.handleOk} {...others} >
                <p>{Translate.tr('notificationDate')}</p>
                <DatePicker onChange={this.handleDateChanged} value={moment(defaultDate)} />
                <p>{Translate.tr('notificationTime')}</p>
                <TimePicker onChange={this.handleTimeChanged} value={moment(defaultTime)} format={'HH:mm'} />
                <p>{Translate.tr('eventDescription')}</p>
                <Input.TextArea defaultValue={defaultDescriptin} onChange={this.handleDescriptionChanged} rows={4} />
            </Modal>
        );
    }
}
