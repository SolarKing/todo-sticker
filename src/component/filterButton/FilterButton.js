// @flow
import React from 'react';
import { Button } from 'antd';
import Translate from '../../class/translate';
import FilterPanel from '../filterPanel/FilterPanel';

type Props = {

}

type States = {
    showPanel: boolean,
}

export default class FilterButton extends React.Component<Props, States> {

    state: States = {
        showPanel: false,
    }

    constructor(props: Props) {
        super(props);
    }

    render() {
        return (
            <div {...this.props}>
                <Button onClick={() => this.setState({showPanel: !this.state.showPanel})} >{Translate.tr('filter')}</Button>
                <FilterPanel show={this.state.showPanel} />
            </div>
        );
    }
}
