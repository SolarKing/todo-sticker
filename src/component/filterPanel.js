// @flow
import React from 'react';
import Translate from '../class/translate';
import { Radio } from 'antd';
import Classnames from 'classnames';
import { CompleteStatusFilterEnum, DueDateFilterEnum} from '../constant/filter';
import type {DueDateFilter, CompleteStatusFilter} from '../constant/filter';
import styled from 'styled-components';

type Props = {
    show: boolean,
    onDueDateFilterChanged: (filter: DueDateFilter) => void,
    onCompleteStatusFilterChanged: (filter: CompleteStatusFilter) => void,
    onHide: () => void,
    defaultDueDateFilter: DueDateFilter,
    defaultCompleteStatusFilter: CompleteStatusFilter,
}

export default class FilterPanel extends React.Component<Props> {

    static defaultProps = {
        show: false
    }

    _handleClickOutside() {
        this.props.onHide();
    }

    _handleDueDateFilterSelected(event: any) {
        let seleted: DueDateFilter = event.target.value;
        this.props.onDueDateFilterChanged(seleted);
    }

    _handleCompleteStatusFilterSelected(event: any) {
        let seleted: CompleteStatusFilter = event.target.value;
        this.props.onCompleteStatusFilterChanged(seleted);
    }

    render() {
        let {
            onDueDateFilterChanged,
            onCompleteStatusFilterChanged,
            show,
            onHide: onHidden,
            defaultDueDateFilter,
            defaultCompleteStatusFilter,
            ...others
        } = this.props;

        let usedClassName = Classnames({
            'close': !show,
        });

        return (
            <div>
                <FilterPanelDiv 
                    {...others}
                    className={usedClassName}
                >
                    <FilterGroupDiv>
                        <p>{Translate.tr('Due Date:')}</p>
                        <Radio.Group 
                            onChange={this._handleDueDateFilterSelected.bind(this)}
                            defaultValue={defaultDueDateFilter}
                        >
                            <Radio.Button
                                value={DueDateFilterEnum.today}
                            >
                                {Translate.tr('Today')}
                            </Radio.Button>
                            <Radio.Button
                                value={DueDateFilterEnum.all}
                            >
                                {Translate.tr('All')}
                            </Radio.Button>
                        </Radio.Group>
                    </FilterGroupDiv>
                    <hr/>
                    <FilterGroupDiv >
                        <p>{Translate.tr('Complete Status:')}</p>
                        <Radio.Group
                            onChange={this._handleCompleteStatusFilterSelected.bind(this)}
                            defaultValue={defaultCompleteStatusFilter}
                        >
                            <Radio.Button 
                                value={CompleteStatusFilterEnum.all}
                            >
                                {Translate.tr('All')}
                            </Radio.Button>
                            <Radio.Button
                                value={CompleteStatusFilterEnum.complete}
                            >
                                {Translate.tr('Complete')}
                            </Radio.Button>
                            <Radio.Button
                                value={CompleteStatusFilterEnum.uncomplete}
                            >
                                {Translate.tr('Uncompleted')}
                            </Radio.Button>
                        </Radio.Group>
                    </FilterGroupDiv>
                </FilterPanelDiv>
                {
                    (show? (<MaskDiv onClick={this._handleClickOutside.bind(this)} />): null)
                }
            </div>
        );
    }

}

const FilterPanelDiv = styled.div`
    position: absolute;
    z-index: 100;
    width: 80%;
    height: auto;
    background-color: #ffffff;
    padding: 10px;
    box-shadow: 0px 0px 3px #888888;
    transition: all .2s ease-in-out;
    transform: scale(1);
    transform-origin: 0% 0%;
    &.close {
        transform: scale(0);
    }
`;
FilterPanelDiv.displayName = 'FilterPanelDiv';

const FilterGroupDiv = styled.div`
    margin: 10px 0;
`;
FilterGroupDiv.displayName = 'FilterGroupDiv';

const MaskDiv = styled.div`
    top: 0;
    left: 0;
    opacity: 0;
    z-index: 99;
    position: fixed;
    width: 100%;
    height: 100%;
`;
MaskDiv.displayName = 'MaskDiv';