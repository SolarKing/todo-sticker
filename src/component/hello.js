// @flow
import React from 'react';
import { remote } from 'electron';
// const { remote } = require('electron')

type Props = {
    name: string
}

export default (props: Props) => (
    <div>Hello {props.name}</div>
)