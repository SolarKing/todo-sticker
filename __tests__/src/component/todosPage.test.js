// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import TodosPage from '../../../src/component/todosPage';
import FilterButton from '../../../src/component/filterButton';
import FilterPanelContainer from '../../../src/container/filterPanelContainer';
import { TodoEditableModal } from '../../../src/component/eventEditModal';

describe('TodosPage', () => {
    let wrapper: ShallowWrapper;
    beforeAll(() => {
        wrapper = shallow(<TodosPage/>);
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should show filter panel', () => {
        wrapper.find(FilterButton).simulate('click');
        expect(wrapper.state().showPanel).toBeTruthy();
    });

    it('should hide filter panel', () => {
        wrapper.find(FilterPanelContainer).simulate('hidden');
        expect(wrapper.state().showPanel).toBeFalsy();
    });

    it('should show editable modal', () => {
        let show = jest.spyOn(TodoEditableModal, 'show');
        wrapper.find('AddButton').simulate('click');
        expect(show.mock.calls.length).toEqual(1);
    });

});