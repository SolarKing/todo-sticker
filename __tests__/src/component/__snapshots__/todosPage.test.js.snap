// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosPage should render correct 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodosPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          <ForwardRef />
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef>
            <FilterButton
              onClick={[Function]}
            />
            <Connect(FilterPanel)
              onHidden={[Function]}
              show={false}
            />
          </ForwardRef>
          <ForwardRef>
            <Connect(TodoList) />
          </ForwardRef>
          <ForwardRef
            style={
              Object {
                "display": "inline",
              }
            }
          />
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "todosPage__AddEventToolButton-sc-2rfgk7-1",
              "isStatic": true,
              "rules": Array [
                "align-self:center;margin-right:10px;",
              ],
            },
            "displayName": "todosPage__AddEventToolButton",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "todosPage__AddEventToolButton-sc-2rfgk7-1",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "todosPage__ToolBar-sc-2rfgk7-2",
            "isStatic": true,
            "rules": Array [
              "grid-area:tool-bar;display:flex;justify-content:flex-end;background-color:#fcfcfc;box-shadow:0px 5px 15px #a5a5a5;",
            ],
          },
          "displayName": "todosPage__ToolBar",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "todosPage__ToolBar-sc-2rfgk7-2",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              <FilterButton
                onClick={[Function]}
              />
              <Connect(FilterPanel)
                onHidden={[Function]}
                show={false}
              />
            </ForwardRef>,
            <ForwardRef>
              <Connect(TodoList) />
            </ForwardRef>,
            <ForwardRef
              style={
                Object {
                  "display": "inline",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <FilterButton
                  onClick={[Function]}
                />,
                <Connect(FilterPanel)
                  onHidden={[Function]}
                  show={false}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onHidden": [Function],
                  "show": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "todosPage__FilterButtonDiv-sc-2rfgk7-4",
                "isStatic": true,
                "rules": Array [
                  "grid-area:filterButton;align-self:end;margin-left:10px;",
                ],
              },
              "displayName": "todosPage__FilterButtonDiv",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "todosPage__FilterButtonDiv-sc-2rfgk7-4",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(TodoList) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "todosPage__TodoListDiv-sc-2rfgk7-5",
                "isStatic": true,
                "rules": Array [
                  "grid-area:listArea;",
                ],
              },
              "displayName": "todosPage__TodoListDiv",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "todosPage__TodoListDiv-sc-2rfgk7-5",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "style": Object {
                "display": "inline",
              },
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "WrappedComponent": [Function],
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "todosPage__SortButtonDiv-sc-2rfgk7-6",
                "isStatic": true,
                "rules": Array [
                  "grid-area:sort-btn;justify-self:end;align-self:end;margin-right:10px;",
                ],
              },
              "displayName": "todosPage__SortButtonDiv",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "todosPage__SortButtonDiv-sc-2rfgk7-6",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "todosPage__TodosView-sc-2rfgk7-3",
            "isStatic": true,
            "rules": Array [
              "grid-area:todos-view;display:grid;width:95%;height:90%;justify-self:center;align-self:start;grid-row-gap:10px;grid-template-columns:10% 70% 20%;grid-template-rows:10% 90%;grid-template-areas:\\"filterButton sort-btn sort-btn\\" \\"listArea listArea listArea\\";background-color:#fcfcfc;",
            ],
          },
          "displayName": "todosPage__TodosView",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "todosPage__TodosView-sc-2rfgk7-3",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "todosPage__TodosContainer-sc-2rfgk7-0",
        "isStatic": true,
        "rules": Array [
          "width:100%;height:100vh;display:grid;grid-template-columns:10% 70% 20%;grid-template-rows:10% 90%;grid-row-gap:5%;grid-template-areas:\\"tool-bar tool-bar tool-bar\\" \\"todos-view todos-view todos-view\\";background-color:#d8d8d8;",
        ],
      },
      "displayName": "todosPage__TodosContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "todosPage__TodosContainer-sc-2rfgk7-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            <ForwardRef />
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef>
              <FilterButton
                onClick={[Function]}
              />
              <Connect(FilterPanel)
                onHidden={[Function]}
                show={false}
              />
            </ForwardRef>
            <ForwardRef>
              <Connect(TodoList) />
            </ForwardRef>
            <ForwardRef
              style={
                Object {
                  "display": "inline",
                }
              }
            />
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "todosPage__AddEventToolButton-sc-2rfgk7-1",
                "isStatic": true,
                "rules": Array [
                  "align-self:center;margin-right:10px;",
                ],
              },
              "displayName": "todosPage__AddEventToolButton",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "todosPage__AddEventToolButton-sc-2rfgk7-1",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "todosPage__ToolBar-sc-2rfgk7-2",
              "isStatic": true,
              "rules": Array [
                "grid-area:tool-bar;display:flex;justify-content:flex-end;background-color:#fcfcfc;box-shadow:0px 5px 15px #a5a5a5;",
              ],
            },
            "displayName": "todosPage__ToolBar",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "todosPage__ToolBar-sc-2rfgk7-2",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef>
                <FilterButton
                  onClick={[Function]}
                />
                <Connect(FilterPanel)
                  onHidden={[Function]}
                  show={false}
                />
              </ForwardRef>,
              <ForwardRef>
                <Connect(TodoList) />
              </ForwardRef>,
              <ForwardRef
                style={
                  Object {
                    "display": "inline",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <FilterButton
                    onClick={[Function]}
                  />,
                  <Connect(FilterPanel)
                    onHidden={[Function]}
                    show={false}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onHidden": [Function],
                    "show": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "todosPage__FilterButtonDiv-sc-2rfgk7-4",
                  "isStatic": true,
                  "rules": Array [
                    "grid-area:filterButton;align-self:end;margin-left:10px;",
                  ],
                },
                "displayName": "todosPage__FilterButtonDiv",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "todosPage__FilterButtonDiv-sc-2rfgk7-4",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Connect(TodoList) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "todosPage__TodoListDiv-sc-2rfgk7-5",
                  "isStatic": true,
                  "rules": Array [
                    "grid-area:listArea;",
                  ],
                },
                "displayName": "todosPage__TodoListDiv",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "todosPage__TodoListDiv-sc-2rfgk7-5",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "style": Object {
                  "display": "inline",
                },
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "WrappedComponent": [Function],
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "todosPage__SortButtonDiv-sc-2rfgk7-6",
                  "isStatic": true,
                  "rules": Array [
                    "grid-area:sort-btn;justify-self:end;align-self:end;margin-right:10px;",
                  ],
                },
                "displayName": "todosPage__SortButtonDiv",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "todosPage__SortButtonDiv-sc-2rfgk7-6",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "todosPage__TodosView-sc-2rfgk7-3",
              "isStatic": true,
              "rules": Array [
                "grid-area:todos-view;display:grid;width:95%;height:90%;justify-self:center;align-self:start;grid-row-gap:10px;grid-template-columns:10% 70% 20%;grid-template-rows:10% 90%;grid-template-areas:\\"filterButton sort-btn sort-btn\\" \\"listArea listArea listArea\\";background-color:#fcfcfc;",
              ],
            },
            "displayName": "todosPage__TodosView",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "todosPage__TodosView-sc-2rfgk7-3",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "todosPage__TodosContainer-sc-2rfgk7-0",
          "isStatic": true,
          "rules": Array [
            "width:100%;height:100vh;display:grid;grid-template-columns:10% 70% 20%;grid-template-rows:10% 90%;grid-row-gap:5%;grid-template-areas:\\"tool-bar tool-bar tool-bar\\" \\"todos-view todos-view todos-view\\";background-color:#d8d8d8;",
          ],
        },
        "displayName": "todosPage__TodosContainer",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "todosPage__TodosContainer-sc-2rfgk7-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
